### PR TITLE
make the transaction sending more transparent

### PR DESCRIPTION
**Pull request by jarry-xiao**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -414,61 +415,70 @@ impl ClientSubsetSync for RpcClient {
...
-                    }
+        let blockhash = self
+            .get_latest_blockhash_with_commitment(CommitmentConfig::confirmed())?
+            .0;
+        tx.partial_sign(&signers.to_vec(), blockhash);
+        let signature = self.send_transaction_with_config(
+            &tx,
+            RpcSendTransactionConfig {
+                skip_preflight: false,
+                preflight_commitment: Some(CommitmentLevel::Processed),
```
Comment (eugene-chen) [inline src/lib.rs]: do you want to keep this?

File: src/lib.rs [hunk 2]
```diff
@@ -414,61 +415,70 @@ impl ClientSubsetSync for RpcClient {
...
+                encoding: None,
+                max_retries: None,
+                min_context_slot: None,
+            },
+        )?;
+
+        let (signature, status) = loop {
+            // Get recent commitment in order to count confirmations for successful transactions
+            let status = self
+                .get_signature_status_with_commitment(&signature, CommitmentConfig::processed())?;
```
Comment (eugene-chen) [inline src/lib.rs:436]: processed?

File: src/lib.rs [hunk 3]
```diff
@@ -414,61 +415,70 @@ impl ClientSubsetSync for RpcClient {
...
+                    &tx.message.recent_blockhash,
+                    CommitmentConfig::processed(),
+                )?;
+                if blockhash_not_found {
+                    break (signature, status);
                 }
+            } else {
+                break (signature, status);
             }
+            sleep(Duration::from_millis(200));
```
Comment (eugene-chen) [inline src/lib.rs]: this is a lot for sync sleep

**Review by eugene-chen — APPROVED** [review]

File: src/lib.rs (hunk 2 above)
Comment (jarry-xiao) [reply src/lib.rs:436]: this first step makes sure that someone received our tx

File: src/lib.rs (hunk 3 above)
Comment (jarry-xiao) [reply src/lib.rs]: 50 then?

**Pull request merged**